test/Microsoft.AspNet.Routing.Tests/Template/TemplateParserTests.cs

```diff
@@ -462,7 +462,33 @@ public void Parse_ComplexSegment_OptionalParameterFollowingPeriod_PeriodAfterSla
                                                                         null,
                                                                         null));
             expected.Parameters.Add(expected.Segments[0].Parts[0]);
-            expected.Parameters.Add(expected.Segments[1].Parts[1]);            
+            expected.Parameters.Add(expected.Segments[1].Parts[1]);
+
+            // Act
+            var actual = TemplateParser.Parse(template);
+
+            // Assert
+            Assert.Equal<RouteTemplate>(expected, actual, new TemplateEqualityComparer());
+        }
+
+        [Theory]
+        [InlineData(@"{p1:regex(^\d{{3}}-\d{{3}}-\d{{4}}$)}")] // ssn
+        [InlineData(@"{p1:regex(^\d{{1,2}}\/\d{{1,2}}\/\d{{4}}$)}")] // date
+        [InlineData(@"{p1:regex(^\w+\@\w+\.\w+)}")] // email
+        [InlineData(@"{p1:regex(([}}])\w+}")] // Not balanced }
+        [InlineData(@"{p1:regex(([{{(])\w+}")] // Not balanced }
+        public void Parse_RegularExpressions(string template)
+        {
+            // Arrange
+            var expected = new RouteTemplate(new List<TemplateSegment>());
+            expected.Segments.Add(new TemplateSegment());
+            InlineConstraint c = new InlineConstraint(template);
```

Comment:
This doesn't seem right. The inline constraint shouldn't have ALL the text. It should just have `:regex(([{{(])\w+` (or similar)